### ISSUE TITLE
[DML] [iGPU] [AMD] GPU command exception

### ISSUE THREAD

**Issue by tjtanaa** [microsoft/onnxruntime-genai#667]
I encounter the following error:
```
2024-07-02 17:22:22.977 | INFO     | embeddedllm.engine:generate:359 - Generation Request Prompt length: 970, New tokens: 50
2024-07-02 17:22:26.036 | ERROR    | embeddedllm.engine:generate:432 - D:\a\_work\1\onnxruntime-genai\src\dml\dml_command_recorder.cpp(143)\onnxruntime_genai.cp310-win_amd64.pyd!00007FFAA661D3A3: (caller: 00007FFAA66112A5) Exception(1) tid(638e0) 887A0006 The GPU will not respond to more commands, most likely because of an invalid command passed by the calling application.
```

## Platform
1. AMD Radeon 780M Graphics
2. 8GB Dedicated GPU memory
3. 27.9GB Shared GPU memory
4. Model: https://huggingface.co/microsoft/Phi-3-mini-4k-instruct-onnx/tree/main/directml/directml-int4-awq-block-128 

## Step to reproduce

1. `conda create -n onnxdml python=3.10`
2. `conda activate onnxdml`
4. `pip install onnxruntime-genai-directml`
5. Follow the example at https://onnxruntime.ai/docs/genai/tutorials/phi3-python.html#run-with-directml .

Params:
```
do_sample: True
temperature: 0.1
top_k: 1,
top_p: 0.1
output_tokens: 50
```

Input Prompt
```
Summarize the following text:  Sea turtles superfamily Chelonioidea sometimes called marine turtles3 are reptiles of the order Testudines and of the suborder Cryptodira The seven existing species of sea turtles are the flatback green hawksbill leatherback loggerhead Kemps ridley and olive ridley45 Six of the seven sea turtle species all but the flatback are present in US waters and are listed as endangered andor threatened under the Endangered Species Act6 All but the flatback turtle are listed as threatened with extinction globally on the IUCN Red List of Threatened Species The flatback turtle is found only in the waters of Australia Papua New Guinea and Indonesia76  Sea turtles can be categorized as hardshelled cheloniid or leatheryshelled dermochelyid8 The only dermochelyid species of sea turtle is the leatherback8  Description For each of the seven species of sea turtles females and males are the same size As adults it is possible to tell male turtles from female turtles by their long tails with a cloacal opening near the tip Adult female sea turtles have shorter tails with a cloacal opening near the base Hatchling and subadult turtles do not exhibit sexual dimorphism it is not possible to determine their sex by looking at them9  In general sea turtles have a more fusiform body plan than their terrestrial or freshwater counterparts This tapering at both ends reduces volume and means that sea turtles cannot retract their head and limbs into their shells for protection unlike many other turtles and tortoises10 However the streamlined body plan reduces friction and drag in the water and allows sea turtles to swim more easily and swiftly  The leatherback sea turtle is the largest sea turtle reaching 14 to more than 18 m 46 to 59 ft in length and weighing between 300 and 640 kg 661 to 1411 lbs11 Other sea turtle species are smaller ranging from as little as 60 cm 2 ft long in the case of the Kemps ridley which is the smallest sea turtle species to 120 cm 39 ft long in the case of the green turtle the second largest512  The skulls of sea turtles have cheek regions that are enclosed in bone1314 Although this condition appears to resemble that found in the earliest known fossil reptiles anapsids it is possible it is a more recently evolved trait in sea turtles placing them outside the anapsids1513  Taxonomy and evolution Sea turtles along with other turtles and tortoises are part of the order Testudines All species except the leatherback sea turtle are in the family Cheloniidae The superfamily name Chelonioidea and family name Cheloniidae are based on the Ancient Greek word for tortoise  khelone16 The leatherback sea turtle is the only extant member of the family Dermochelyidae  Fossil evidence of marine turtles goes back to the Late Jurassic 150 million years ago with genera such as Plesiochelys from Europe In Africa the first marine turtle is Angolachelys from the Turonian of Angola17 A lineage of unrelated marine testudines the pleurodire sidenecked bothremydids also survived well into the Cenozoic Other pleurodires are also thought to have lived at sea such as Araripemys18 and extinct pelomedusids19 Modern sea turtles are not descended from more than one of the groups of seagoing turtles that have existed in the past they instead constitute a single radiation that became distinct from all other turtles at least 110 million years ago202122 Their closest extant relatives are in fact the snapping turtles Chelydridae musk turtles Kinosternidae and hickatee Dermatemyidae of the Americas which alongside the sea turtles constitute the clade Americhelydia23
```

**Comment by tjtanaa** [comment]
> Thank you for reporting this issue. We are looking into it.

Hi @natke , may I know is there any updates? What could be the source of issue? I would like to attempt to look into it if there is any possible sources of issue.

**Comment by elephantpanda** [comment]
Hi @tjtanaa . Do you think this is the same as [my bug here](https://github.com/microsoft/onnxruntime-genai/issues/833). My bug it happens only with certain prompts and not others (but i can't see any pattern to which ones). Is this the same as your bug or does your bug occur with all prompts or only certain ones too?

**Comment by natke** [comment]
Possible duplicate of #833

**Comment by tjtanaa** [comment]
> Hi @tjtanaa . Do you think this is the same as [my bug here](https://github.com/microsoft/onnxruntime-genai/issues/833). My bug it happens only with certain prompts and not others (but i can't see any pattern to which ones). Is this the same as your bug or does your bug occur with all prompts or only certain ones too?

Hi @elephantpanda 
I am not sure if that is the case. Because it fails very quickly with longer input prompt.

**Comment by tjtanaa** [comment]
@elephantpanda with v0.4.0, The input that I shared in this issue has been resolved. I think the issue that I had is about long context length. 

However, new issue arises. The model is outputing gibberish after output more than a few hundred tokens:


The input I passed:
```

    url = "http://localhost:5555/v1/chat/completions"
    payload = {
        "messages": [{"role": "user", "content": "Summarize the following content: Alices Adventures in Wonderland                  ALICES ADVENTURES IN WONDERLAND                            Lewis Carroll                 THE MILLENNIUM FULCRUM EDITION 30                                 CHAPTER I                        Down the RabbitHole     Alice was beginning to get very tired of sitting by her sister on the bank and of having nothing to do  once or twice she had peeped into the book her sister was reading but it had no pictures or conversations in it and what is the use of a book thought Alice without pictures or conversation    So she was considering in her own mind as well as she could for the hot day made her feel very sleepy and stupid whether the pleasure of making a daisychain would be worth the trouble of getting up and picking the daisies when suddenly a White Rabbit with pink eyes ran close by her    There was nothing so VERY remarkable in that nor did Alice think it so VERY much out of the way to hear the Rabbit say to itself Oh dear  Oh dear  I shall be late  when she thought it over afterwards it occurred to her that she ought to have wondered at this but at the time it all seemed quite natural but when the Rabbit actually TOOK A WATCH OUT OF ITS WAISTCOAT POCKET and looked at it and then hurried on Alice started to her feet for it flashed across her mind that she had never before seen a rabbit with either a waistcoatpocket or a watch to take out of it and burning with curiosity she ran across the field after it and fortunately was just in time to see it pop down a large rabbithole under the hedge    In another moment down went Alice after it never once considering how in the world she was to get out again    The rabbithole went straight on like a tunnel for some way and then dipped suddenly down so suddenly that Alice had not a moment to think about stopping herself before she found herself falling down a very deep well    Either the well was very deep or she fell very slowly for she had plenty of time as she went down to look about her and to wonder what was going to happen next  First she tried to look down and make out what she was coming to but it was too dark to see anything then she looked at the sides of the well and noticed that they were filled with cupboards and bookshelves here and there she saw maps and pictures hung upon pegs  She took down a jar from one of the shelves as she passed it was labelled ORANGE MARMALADE but to her great disappointment it was empty  she did not like to drop the jar for fear of killing somebody so managed to put it into one of the cupboards as she fell past it    Well thought Alice to herself after such a fall as this I shall think nothing of tumbling down stairs  How brave theyll all think me at home  Why I wouldnt say anything about it even if I fell off the top of the house Which was very likely true    Down down down  Would the fall NEVER come to an end  I wonder how many miles Ive fallen by this time she said aloud I must be getting somewhere near the centre of the earth  Let me see  that would be four thousand miles down I think for you see Alice had learnt several things of this sort in her lessons in the schoolroom and though this was not a VERY good opportunity for showing off her knowledge as there was no one to listen to her still it was good practice to say it over yes thats about the right distancebut then I wonder what Latitude or Longitude Ive got to  Alice had no idea what Latitude was or Longitude either but thought they were nice grand words to say    Presently she began again  I wonder if I shall fall right THROUGH the earth  How funny itll seem to come out among the people that walk with their heads downward  The Antipathies I think she was rather glad there WAS no one listening this time as it didnt sound at all the right word but I shall have to ask them what the name of the country is you know Please Maam is this New Zealand or Australia and she tried to curtsey as she spokefancy CURTSEYING as youre falling through the air  Do you think you could manage it  And what an ignorant little girl shell think me for asking  No itll never do to ask  perhaps I shall see it written up somewhere    Down down down  There was nothing else to do so Alice soon began talking again  Dinahll miss me very much tonight I should think  Dinah was the cat  I hope theyll remember her saucer of milk at teatime  Dinah my dear  I wish you were down here with me  There are no mice in the air Im afraid but you might catch a bat and thats very like a mouse you know But do cats eat bats I wonder  And here Alice began to get rather sleepy and went on saying to herself in a dreamy sort of way Do cats eat bats  Do cats eat bats and sometimes Do bats eat cats for you see as she couldnt answer either question it didnt much matter which way she put it  She felt that she was dozing off and had just begun to dream that she was walking hand in hand with Dinah and saying to her very earnestly Now Dinah tell me the truth  did you ever eat a bat when suddenly thump thump down she came upon a heap of sticks and dry leaves and the fall was over    Alice was not a bit hurt and she jumped up on to her feet in a moment  she looked up but it was all dark overhead before her was another long passage and the White Rabbit was still in sight hurrying down it  There was not a moment to be lost away went Alice like the wind and was just in time to hear it say as it turned a corner Oh my ears and whiskers how late its getting  She was close behind it when she turned the corner but the Rabbit was no longer to be seen  she found herself in a long low hall which was lit up by a row of lamps hanging from the roof    There were doors all round the hall but they were all locked and when Alice had been all the way down one side and up the other trying every door she walked sadly down the middle wondering how she was ever to get out again    Suddenly she came upon a little threelegged table all made of solid glass there was nothing on it except a tiny golden key and Alices first thought was that it might belong to one of the doors of the hall but alas either the locks were too large or the key was too small but at any rate it would not open any of them  However on the second time round she came upon a low curtain she had not noticed before and behind it was a little door about fifteen inches high  she tried the little golden key in the lock and to her great delight it fitted    Alice opened the door and found that it led into a small passage not much larger than a rathole  she knelt down and looked along the passage into the loveliest garden you ever saw How she longed to get out of that dark hall and wander about among those beds of bright flowers and those cool fountains but she could not even get her head through the doorway and even if my head would go through thought poor Alice it would be of very little use without my shoulders  Oh how I wish I could shut up like a telescope  I think I could if I only know how to begin  For you see so many outoftheway things had happened lately that Alice had begun to think that very few things indeed were really impossible    There seemed to be no use in waiting by the little door so she went back to the table half hoping she might find another key on it or at any rate a book of rules for shutting people up like telescopes  this time she found a little bottle on it which certainly was not here before said Alice and round the neck of the bottle was a paper label with the words DRINK ME beautifully printed on it in large letters    It was all very well to say Drink me but the wise little Alice was not going to do THAT in a hurry  No Ill look first she said and see whether its marked poison or not for she had read several nice little histories about children who had got burnt and eaten up by wild beasts and other unpleasant things all because they WOULD not remember the simple rules their friends had taught them  such as that a redhot poker will burn you if you hold it too long and that if you cut your finger VERY deeply with a knife it usually bleeds and she had never forgotten that if you drink much from a bottle marked poison it is almost certain to disagree with you sooner or later    However this bottle was NOT marked poison so Alice ventured to taste it and finding it very nice it had in fact a sort of mixed flavour of cherrytart custard pineapple roast turkey toffee and hot buttered toast she very soon finished it off                                                                                                                                                    What a curious feeling said Alice I must be shutting up like a telescope"}],
        "model": "phi3-mini-int4",
        "max_tokens": 1000,
        "temperature": 0.0,
        "stream": True,
    }
```


Generation Output:
```
Alice'slife in Wonderland: A Tale

Alice's Adventures in Wonderland is a novel written by Lewis Caroll. The story follows the adventures of Alice, a young girl, who falls down a rabbit hole and enters a fantastical world. In the book, Alice encounters various characters, including a White Rabbit, a Mad Hatter, and a Queen of Hearts.

The book was first published in 1865 and is considered a classic in children's literature. It has been adapted into various forms of media, including a 2010 film.

Summary:

Alice's Adventures in Wonderland is a tale of a young girl's journey through a magical world. The story begins with Alice, who is bored and restless, sitting by the riverbank. She spots a White Rabbit with its large teeth and decides to follow it. Alice follows the White Rabbit down a rabbit hole, and finds herself in a strange and bizarre world.

In this world, Alice meets a Mad Hatter, a March Hare, and a variety of other peculiar characters. She also encounters a Mad Hatter, a March Hare, and a Hatter. The Mad Hatter, Alice in her world.

Alice soon finds herself in a nonsensical world. She soon finds herself in a nonsensical world. Alice in Wonderland. Alice soon finds herself in a nonsensical world. Alice soon finds herself in a nonsensical world.

Alice's Adventures in Wonderland Alices Adventures in Wonderland                        Adventures in Wonderland    
                    Alice's Adventures in Wonderland                        Alice in Wonderland                     
   Adventures in Wonderland                        Alice's Adventures in Wonderland                        Alice in  It                        Alice'


 Alice's AdventURESonoracle began to read allormans.

```
